airflow/models/taskinstance.py

```diff
@@ -1265,6 +1558,132 @@ def _update_rtif(ti, rendered_fields, session: Session | None = None):
     RenderedTaskInstanceFields.delete_old_records(ti.task_id, ti.dag_id, session=session)
 
 
+def _coalesce_to_orm_ti(*, ti: TaskInstancePydantic | TaskInstance, session: Session):
+    from airflow.models.dagrun import DagRun
+    from airflow.serialization.pydantic.taskinstance import TaskInstancePydantic
+
+    if isinstance(ti, TaskInstancePydantic):
+        orm_ti = DagRun.fetch_task_instance(
+            dag_id=ti.dag_id,
+            dag_run_id=ti.run_id,
+            task_id=ti.task_id,
+            map_index=ti.map_index,
+            session=session,
+        )
+        if TYPE_CHECKING:
+            assert orm_ti
+        ti, pydantic_ti = orm_ti, ti
+        _set_ti_attrs(ti, pydantic_ti)
+        ti.task = pydantic_ti.task
+    return ti
+
+
+@internal_api_call
+@provide_session
+def _defer_task(
```

Comment:
as above - make naming distinct.
```suggestion
def _defer_ti_task(
```